src/frontend/components/PictureInPictureLayer/usePIPDragger/index.tsx

```diff
@@ -0,0 +1,223 @@
+      return;
+    }
+
+    if (initialDragPoint.current === undefined) {
+      //  picture is not currently dragged
+      return;
+    }
+
+    initialDragPoint.current = undefined;
+
+    updatePIPPosition(
+      computePicturePosition(
+        velocityRef.current,
+        backgroundRef.current,
+        foregroundRef.current,
+        defaultmarginSize,
+      ),
+      true,
+    );
+  }, [updatePIPPosition]);
+  const startDragging = useCallback(
+    (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      if (foregroundRef.current === null || backgroundRef.current === null) {
+        return;
+      }
+
+      initialDragPoint.current = { x: event.clientX, y: event.clientY };
+
+      document.addEventListener('mouseup', endDragging);
+      document.addEventListener('mousemove', drag);
+    },
+    [drag, endDragging],
+  );
+
+  //  update picture position on container resize
+  useEffect(() => {
+    if (!container) {
+      return;
+    }
+
+    const observer = new ResizeObserver(() => {
+      if (!backgroundRef.current || !foregroundRef.current) {
+        return;
+      }
+
+      updatePIPPosition(
+        computePicturePosition(
+          velocityRef.current,
+          backgroundRef.current,
+          foregroundRef.current,
+          defaultmarginSize,
+        ),
+        true,
+      );
+    });
+    observer.observe(container);
+
+    return () => {
+      if (!container) {
+        return;
+      }
+
+      observer.unobserve(container);
+    };
+  }, [container]);
+
+  useEffect(() => {
+    if (firstElementRef.current === null || !reversed) {
+      return;
+    }
+
+    const observer = new ResizeObserver(() => {
+      if (!backgroundRef.current || !foregroundRef.current) {
+        return;
+      }
+
+      updatePIPPosition(
+        computePicturePosition(
+          velocityRef.current,
+          backgroundRef.current,
+          foregroundRef.current,
+          defaultmarginSize,
+        ),
+        true,
+      );
+    });
+    observer.observe(firstElementRef.current);
+
+    return () => {
+      if (!firstElementRef.current) {
+        return;
+      }
+
+      observer.unobserve(firstElementRef.current);
+    };
+  }, [reversed]);
+
+  useEffect(() => {
+    if (secondElementRef.current === null || reversed) {
+      return;
+    }
+
+    const observer = new ResizeObserver(() => {
+      if (!backgroundRef.current || !foregroundRef.current) {
+        return;
+      }
+
+      updatePIPPosition(
+        computePicturePosition(
+          velocityRef.current,
+          backgroundRef.current,
+          foregroundRef.current,
+          defaultmarginSize,
+        ),
+        true,
+      );
+    });
+    observer.observe(secondElementRef.current);
+
+    return () => {
+      if (!secondElementRef.current) {
+        return;
+      }
+
+      observer.unobserve(secondElementRef.current);
+    };
+  }, [reversed]);
```

Comment:
IMO, we could use only one useEffect here, and factorize the logic to a function called once with firstElementRef and after with secondElementRef.